generator/src/main/java/com/scottlogic/deg/generator/generation/ChecksummedCodeStringGenerator.java

```diff
@@ -0,0 +1,128 @@
+package com.scottlogic.deg.generator.generation;
+
+import com.scottlogic.deg.generator.utils.*;
+
+import java.util.Arrays;
+import java.util.Objects;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+public abstract class ChecksummedCodeStringGenerator implements StringGenerator {
+    protected final RegexStringGenerator sansCheckDigitGenerator;
+    protected final boolean negate;
+
+    public ChecksummedCodeStringGenerator(String generationPatternSansCheck) {
+        negate = false;
+        sansCheckDigitGenerator = new RegexStringGenerator(generationPatternSansCheck, true);
+    }
+
+    public ChecksummedCodeStringGenerator(RegexStringGenerator generator, boolean negate) {
+        this.negate = negate;
+        sansCheckDigitGenerator = generator;
+    }
+
+    public abstract char calculateCheckDigit(String str);
```

Comment:
Minor comment, then approved can you group all the abstract methods together in this class